common/kernel/timing.cc

```diff
@@ -1216,6 +1392,10 @@ void timing_analysis(Context *ctx, bool print_histogram, bool print_fmax, bool p
                   (update_results && ctx->detailed_timing_report) ? &detailed_net_timings : nullptr);
     timing.walk_paths();
 
+    // Use TimingAnalyser to determine clock-to-clock relations
+    TimingAnalyser timingAnalyser (ctx);
+    timingAnalyser.setup();
```

Comment:
another thought not for this PR but if you have some capacity for general nextpnr development (cc @rowanG077 too) - it would be nice to finally replace all the old timing check code with the new `timingAnalyser` that's already used during placement